mithril-common/src/signed_entity_lock.rs

```diff
@@ -0,0 +1,192 @@
+//! # Signed Entity Lock
+//!
+//! This module provides a non-blocking lock mechanism for signed entities to prevent multiple
+//! threads from modifying the same entity at the same time.
+
+use std::collections::BTreeSet;
+
+use tokio::sync::RwLock;
+
+use crate::entities::SignedEntityTypeDiscriminants;
+
+/// Non-blocking lock mechanism for signed entities to prevent multiple threads from
+/// modifying the same entity at the same time.
```

Comment:
This should probably mention the fact that the runtime will not work on a locked signed entity type, as there is no case yet where the same entity type is accessed by different threads.